homeassistant/core.py

```diff
@@ -166,6 +166,15 @@ class EventStateChangedData(TypedDict):
     new_state: State | None
 
 
+class EventStateReportedData(TypedDict):
+    """EventStateReported data."""
+
+    entity_id: str
+    old_last_reported: NotRequired[datetime.datetime]
+    old_state: NotRequired[State | None]
+    new_state: State | None
+
+
```

Comment:
I was mucking around with this in another change. The reported event is actually a union of changed or reported structures. It was somewhat tricky to get the typing correct.

The EVENT_STATE_REPORTED constant should really be set up go map to the event definition, but was tricky to right with a union of types. Maybe that works with this squashed class.

If we have a squashed class, maybe there should be some doc string to indicate that it contains the full changed event or the reported event?

Reply:
Were not fully view i expected it to be resolved. Somewhat easy to break this if the changed event data is modified. Typing wont catch that reported event piggy backs on the same data as changed.
But im okey with this.